modules/table/src/main/java/org/apache/ignite/internal/table/distributed/raft/snapshot/incoming/IncomingSnapshotCopier.java

```diff
@@ -215,6 +249,28 @@ private CompletableFuture<?> loadSnapshotMeta(ClusterNode snapshotSender) {
         }
     }
 
+    private boolean metadataIsSufficientlyComplete() {
+        return partitionSnapshotStorage.catalogService().latestCatalogVersion() >= snapshotMeta.requiredCatalogVersion();
+    }
+
+    private void logMetadataIsInsufficiencyAndSetError() {
+        LOG.warn(
+                "Metadata not yet available, URI '{}', required level {}; rejecting snapshot installation.",
+                this.snapshotUri,
+                snapshotMeta.requiredCatalogVersion()
+        );
+
+        String errorMessage = String.format(
+                "Metadata not yet available, URI '%s', required level %s; rejecting snapshot installation.",
+                this.snapshotUri,
+                snapshotMeta.requiredCatalogVersion()
+        );
+
+        if (isOk()) {
```

Comment:
Again, I don't quite get it, why do we print warnings/errors if everything's OK. Please comment this code, or fix it if it's wrong

Reply:
It's something completely different. 'If there is no error registered yet, register an error; otherwise leave an earlier error to not overwrite it'. To be honest, I don't see what should be commented here: 'if still no error, raise error status'? But is seems obvious from the code :)